### DISCUSSION TITLE
New feature : no separator

### DISCUSSION THREAD

**Discussion by woosaaahh** [woosaaahh/sj.nvim#22] [Announcements]
Hi everyone !

I just added a new feature here : [`feat-no_separator`](https://github.com/woosaaahh/sj.nvim/tree/feat-no_separator).

The goal of this feature is to avoid typing an extra character for the separator.
The downside is that it might reduces available labels because each label that can match the next character after a match is discarded.

To try this feature, use the `feat-no_separator` branch and set the separator option to an empty string : `separator = ""`.

This feature is not yet merged into `main` or `0.6`, same thing for the multi windows feature.
This means that if you want to use of one the two features, you can't use the other yet.

@jemag & @ziontee113 I'd really appreciate if you could try this feature and give me some feebacks please !

**Comment by ziontee113** [comment]
@woosaaahh OK, this is epic. Feels super natural to use. I'm gonna configure my `sj` config a bit, use it for a while then give more feedback. But so far this feature feels wonderful. Brilliant job :+1:

**Reply by woosaaahh** [reply]
> OK, this is epic. Feels super natural to use.
> But so far this feature feels wonderful. Brilliant job +1

I'm very glad to read that ! Thanks so much !
I'm also very happy if it feels natural for you too.

> I'm gonna configure my sj config a bit, use it for a while then give more feedback.

Nice !

**Comment by woosaaahh** [comment]
I played with the feature for all my nvim sessions and it seems to work pretty well !

I merged the `feat-no_separator` branch in `0.6`.

**Comment by jemag** [comment]
Started playing around with it, so far I do enjoy it. It feels just slightly faster without having to press the separator

**Reply by woosaaahh** [reply]
Nice ! Thank you very much for this feedback !

**Comment by ziontee113** [comment]
Not having the seperator is a big deal. You have the freedom to type as many characters as you want, and you don't have to break the immersion by typing the separator. This is a brilliant implementation. Good job m8 :+1:

**Comment by woosaaahh** [comment]
You both made my day (night actually) with your feedbacks !!!
Honestly I'm so glad that you like it !

I don't know if you noticed but `0.6` has been merged to `main`.
